modules/shared/src/main/scala/io/constellationnetwork/currency/http/Decoders.scala

```diff
@@ -0,0 +1,42 @@
+package io.constellationnetwork.currency.http
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+
+import io.constellationnetwork.currency.dataApplication.DataTransaction.DataTransactions
+import io.constellationnetwork.currency.dataApplication.{DataTransaction, DataUpdate, FeeTransaction}
+import io.constellationnetwork.security.signature.Signed
+
+import derevo.circe.magnolia.decoder
+import derevo.derive
+import io.circe.Decoder
+import org.http4s.circe.jsonOf
+import org.http4s.{DecodeResult, EntityDecoder, InvalidMessageBodyFailure}
+
+object Decoders {
+
+  def feeTransactionRequestDecoder[F[_]: Async, D <: DataUpdate](
```

Comment:
it's a single decoder, there are no more implementations of it, so it can be just next to the defined routes

Reply:
I can't access the implementation if I put in the routes, I need this function to be called on the lifecycle function requestDecoder and in test class

Reply:
oh, okay. I was missing that info. it's okay then :+1: